File: hashing/169.java
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();

        for(int i:nums)
        {
            if(!map.containsKey(i))
            {
                map.put(i,1);
            }
            else 
            {
                int ov = map.get(i);
                int nv = ov+1;
                map.put(i,nv);
            }
        }
         int majorityElement = -1;
        int maxCount = 0;
           for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > maxCount) {
                maxCount = entry.getValue();
                majorityElement = entry.getKey();
            }
        }

        return majorityElement;
       
    }
}

File: 169.cpp
class Solution {
public:
  int majorityElement(vector<int> &nums) {
    int vote = 0, candidate = 0;
    for (int i = 0; i < nums.size(); i++) {
      if (vote == 0) {
        candidate = nums[i];
        vote = 1;
      } else if (nums[i] == candidate) {
        vote++;
      } else if (nums[i] != candidate)
        vote--;
    }
    int count = 0;
    for (int i = 0; i < nums.size(); i++) {
      if (candidate == nums[i]) {
        count++;
      }
    }
    if (count > nums.size() / 2)
      return candidate;
    return 0;
  }
};
